docs/extension-maintainers.md

```diff
@@ -0,0 +1,128 @@
+# PIE for Extension Maintainers
+
+Adding PIE support for your extension is relatively straightforward, and the
+flow is quite similar to adding a regular PHP package into Packagist.
+
+## Add a `composer.json` to your extension
+
+The first step to adding PIE support is adding a `composer.json` to your
+extension repository. Most of the typical fields are the same, with a few
+notable exceptions:
+
+ * The `type` MUST be either `php-ext` for a PHP Module (this will be most
+   extensions), or `php-ext-zend` for a Zend Extension.
+ * An additional `php-ext` section MAY exist.
+ * The Composer package name (i.e. the top level `name` field) for a PIE
+   extension cannot share the same Composer package name as a regular PHP
+   package, even if they have different `type` fields.
```

Comment:
Maybe it's worth mentioning that the package name should follow the usual convention for Composer packages (i.e. `<vendor>/<package>` instead of the "usual" `ext-<extension>`) in addition to being unique across the Composer ecosystem.